packages/serverless/src/awslambda.ts

```diff
@@ -57,6 +60,62 @@ export function init(options: Sentry.NodeOptions = {}): void {
   Sentry.addGlobalEventProcessor(serverlessEventProcessor('AWSLambda'));
 }
 
+/** */
+function tryRequire<T>(taskRoot: string, subdir: string, mod: string): T {
+  const lambdaStylePath = path.resolve(taskRoot, subdir, mod);
+  if (fs.existsSync(lambdaStylePath) || fs.existsSync(`${lambdaStylePath}.js`)) {
+    // Lambda-style path
+    return require(lambdaStylePath);
+  }
+  // Node-style path
+  return require(require.resolve(mod, { paths: [taskRoot, subdir] }));
+}
+
+/** */
+export function tryPatchHandler(taskRoot: string, handlerPath: string): void {
+  type HandlerBag = HandlerModule | Handler | null | undefined;
+  interface HandlerModule {
+    [key: string]: HandlerBag;
+  }
+
+  const handlerDesc = path.basename(handlerPath);
+  const match = handlerDesc.match(/^([^.]*)\.(.*)$/);
+  if (!match) {
+    logger.error(`Bad handler ${handlerDesc}`);
+    return;
+  }
+
+  const [, handlerMod, handlerName] = match;
+
+  let obj: HandlerBag;
+  try {
+    const handlerDir = handlerPath.substring(0, handlerPath.indexOf(handlerDesc));
+    obj = tryRequire(taskRoot, handlerDir, handlerMod);
+  } catch (e) {
+    logger.error(`Cannot require ${handlerPath} in ${taskRoot}`, e);
+    return;
+  }
+
+  let mod: HandlerBag;
+  let functionName: string | undefined;
+  handlerName.split('.').forEach(name => {
+    mod = obj;
+    obj = obj && (obj as HandlerModule)[name];
+    functionName = name;
+  });
```

Comment:
This block always uses the result entry of `split` call to override `mod`, `obj` and `functionName`. Was that on purpose?